plugins/custom-codecs/src/main/java/org/opensearch/index/codec/customcodecs/Lucene95CustomCodec.java

```diff
@@ -55,6 +59,16 @@ public Lucene95CustomCodec(Mode mode, int compressionLevel) {
         this.storedFieldsFormat = new Lucene95CustomStoredFieldsFormat(mode, compressionLevel);
     }
 
+    /**
+     * Creates a new compression codec with the given compression level. We use
+     * lowercase letters when registering the codec so that we remain consistent with
+     * the other compression codecs: default, lucene_default, and best_compression.
+     *
+     * @param mode The compression codec (ZSTD or ZSTDNODICT).
+     * @param compressionLevel The compression level.
+     * @param mapperService The mapper service.
+     * @param logger The logger.
+     */
     public Lucene95CustomCodec(Mode mode, int compressionLevel, MapperService mapperService, Logger logger) {
         super("Lucene95CustomCodec", new PerFieldMappingPostingFormatCodec(Lucene95Codec.Mode.BEST_SPEED, mapperService, logger));
```

Comment:
@nknize @sarthakaggarwal97 so here is the issue:
 - Codec is NamedSPI and service discovery should wort for it
 - the problem is (as @sarthakaggarwal97 pointed rightly), the `Lucene95CustomCodec` is used for both codecs
 - we need to add the codec friendly name to the `Mode` enum members (as @sarthakaggarwal97 , `std_no_dict` is not codec friendly)
 - with that, Codec.getAvaliableCodecs() discovery works out of the box, no need to hack the core

```
public Lucene95CustomCodec(Mode mode, int compressionLevel, MapperService mapperService, Logger logger) {
super(mode.codec(), new PerFieldMappingPostingFormatCodec(Lucene95Codec.Mode.BEST_SPEED, mapperService, logger));
}
```

Reply:
What to do with `std_no_dict`? We could introduce new interface called `CodecAliases`:

```
interface CodecAliases {
    String[] aliases();
}
```

That ZstdNoDict woud implement returning `std_no_dict`, the Codec.getAvaliableCodecs() would need to be enhanced with aliases check (we did very similar thing with `CodecSettings` )

Reply:
@reta I'm guessing we will have to iterate through codecs, and based on the implementation of the interface, we would be able to make a decision

Reply:
Correct, but only when there is no match by name, we sadly have to support `std_no_dict` (without hacking the core)